Index: b/readers/supported_readers.txt
===================================================================
--- a/readers/supported_readers.txt
+++ b/readers/supported_readers.txt
@@ -10,6 +10,9 @@
 # section: supported
 ##########################
 
+#WoTrus
+0x4D4B:0x1002:MeSince MK-1
+
 # ACS
 0x072F:0x90CC:ACS ACR 38U-CCID
 
Index: b/src/ccid.c
===================================================================
--- a/src/ccid.c
+++ b/src/ccid.c
@@ -61,6 +61,7 @@ int ccid_open_hack_pre(unsigned int read
 			ccid_descriptor->dwMaxIFSD = 254;
 			break;
 
+		case MESINCE_MK:
 		case CL1356D:
 			/* the firmware needs some time to initialize */
 			(void)sleep(1);
Index: b/src/ccid.h
===================================================================
--- a/src/ccid.h
+++ b/src/ccid.h
@@ -235,6 +235,7 @@ typedef struct
 #define HID_OMNIKEY_6121 0x076B6632 /* OMNIKEY 6121 */
 #define CHERRY_XX44	0x046A00A7 /* Cherry Smart Terminal xx44 */
 #define FUJITSU_D323 0x0BF81024 /* Fujitsu Smartcard Reader D323 */
+#define MESINCE_MK  0x4D4B1002 /* Mesince MK-1 */
 #define IDENTIV_uTrust3700F		0x04E65790
 #define IDENTIV_uTrust3701F		0x04E65791
 #define IDENTIV_uTrust4701F		0x04E65724
